src/riakc_ts_query_operator.erl

```diff
@@ -31,9 +31,12 @@
 -export([serialize/2,
          deserialize/1, deserialize/2]).
 
-serialize(QueryText, Interpolations) ->
+
+serialize(QueryText, Interpolations)
+  when is_binary(QueryText) orelse is_list(QueryText) ->
+    Q = unicode:characters_to_binary(QueryText),
```

Comment:
The original binary passed to serialize should have included the /utf8 qualification. You can’t (or at least shouldn’t) try to force it in here.

-John

> On Aug 2, 2016, at 1:43 PM, Andrei Zavada <notifications@github.com> wrote:
> 
> In src/riakc_ts_query_operator.erl <https://github.com/basho/riak-erlang-client/pull/307#discussion_r73202240>:
> 
> > @@ -31,9 +31,12 @@
> >  -export([serialize/2,
> >           deserialize/1, deserialize/2]).
> >  
> > -serialize(QueryText, Interpolations) ->
> > +
> > +serialize(QueryText, Interpolations)
> > +  when is_binary(QueryText) orelse is_list(QueryText) ->
> > +    Q = unicode:characters_to_binary(QueryText),
> It's tricky. What happens when the QueryText contains <<"фафа">>? unicode:characters_to_binary will not convert it to unicode unless you qualify it with /utf8. Strings, however, are converted as intended.
> 
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub <https://github.com/basho/riak-erlang-client/pull/307/files/fbab1d5e6c8d262977eae400c1366a77757f08d5#r73202240>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ACCsXhnIS__zCs7xpMTNgxbPVqXVa9ILks5qb4HAgaJpZM4Ja1ox>.
>